diff --git a/src/draw/sw/blend/lv_draw_sw_blend_to_i1.c b/src/draw/sw/blend/lv_draw_sw_blend_to_i1.c
index 1fd19c3e7..14c5873c3 100644
--- a/src/draw/sw/blend/lv_draw_sw_blend_to_i1.c
+++ b/src/draw/sw/blend/lv_draw_sw_blend_to_i1.c
@@ -82,7 +82,7 @@ static inline void * /* LV_ATTRIBUTE_FAST_MEM */ drawbuf_next_row(const void * b
  *      MACROS
  **********************/
 
-#define I1_LUM_THRESHOLD 100
+#define I1_LUM_THRESHOLD 127
 
 #ifndef LV_DRAW_SW_I1_BLEND_NORMAL_TO_I1
     #define LV_DRAW_SW_I1_BLEND_NORMAL_TO_I1(...)                    LV_RESULT_INVALID
