diff --git a/modules/lvgl/lvgl_display_mono.c b/modules/lvgl/lvgl_display_mono.c
index 09b9bfd2337..8987659f8bb 100644
--- a/modules/lvgl/lvgl_display_mono.c
+++ b/modules/lvgl/lvgl_display_mono.c
@@ -10,6 +10,9 @@
 #include "lvgl_display.h"
 
 #define COLOR_PALETTE_HEADER_SIZE (8)
+#ifdef CONFIG_PIXEL_FORMAT_MONO10
+#define PIXEL_FORMAT_MONO10 1
+#endif
 
 static uint8_t *mono_conv_buf;
 static uint32_t mono_conv_buf_size;
