diff --git a/modules/lvgl/lvgl_display_mono.c b/modules/lvgl/lvgl_display_mono.c
index c843cd96b54..eca858e041a 100644
--- a/modules/lvgl/lvgl_display_mono.c
+++ b/modules/lvgl/lvgl_display_mono.c
@@ -9,6 +9,9 @@
 #include <string.h>
 #include "lvgl_display.h"
 
+#define COLOR_PALETTE_HEADER_SIZE (8)
+#define PIXEL_FORMAT_MONO10 1
+
 static uint8_t *mono_conv_buf;
 static uint32_t mono_conv_buf_size;
 
@@ -50,8 +53,9 @@ static void lvgl_transform_buffer(uint8_t **px_map, uint32_t width, uint32_t hei
 
 	memset(mono_conv_buf, clear_color, mono_conv_buf_size);
 
-	/* Needed because LVGL reserves 2x4 bytes in the buffer for the color palette. */
-	*px_map += 8;
+	/* Needed because LVGL reserves some bytes in the buffer for the color palette. */
+	*px_map += COLOR_PALETTE_HEADER_SIZE;
+
 	uint8_t *src_buf = *px_map;
 	uint32_t stride = (width + CONFIG_LV_DRAW_BUF_STRIDE_ALIGN - 1) &
 			  ~(CONFIG_LV_DRAW_BUF_STRIDE_ALIGN - 1);
@@ -67,7 +71,7 @@ static void lvgl_transform_buffer(uint8_t **px_map, uint32_t width, uint32_t hei
 		}
 	}
 
-	memcpy(src_buf, mono_conv_buf, mono_conv_buf_size);
+	memcpy(src_buf, mono_conv_buf, mono_conv_buf_size - COLOR_PALETTE_HEADER_SIZE);
 }
 
 void lvgl_flush_cb_mono(lv_display_t *display, const lv_area_t *area, uint8_t *px_map)
